### PR TITLE
Added client validation for creating new channel

### PR DESCRIPTION
**Pull request by Faraser**

### CODE REVIEW

File: app/components/InfoMessage/index.jsx [hunk 1]
```diff
@@ -71,8 +68,8 @@ export default class InfoMessage extends React.Component {
       <div
         className='info-message__text'
         style={{transform: `translateX(-${interpolated.x}%)`}}
-      >
-        {this.state.children}
+        >
```
Comment (Nitive) [inline app/components/InfoMessage/index.jsx]: You are bad guy 💂🏿

File: app/components/InfoMessage/index.jsx [hunk 2]
```diff
@@ -81,7 +78,7 @@ export default class InfoMessage extends React.Component {
         <Motion
           defaultStyle={{x: spring(0)}}
           style={{x: spring(this.state.changing ? 100 : 0, [1000, 100])}}
-        >
+          >
```
Comment (Nitive) [inline app/components/InfoMessage/index.jsx]: You doubly bad guy

File: app/components/NewChannelItem/index.jsx [hunk 3]
```diff
@@ -24,19 +28,28 @@ export default class ChannelItem extends React.Component {
...
     const channelName = this.state.channelName.trim();
-    const sameChannel = this.props.channels.find(channel => channel.get('name') === channelName);
-    if (channelName && !sameChannel) {
-      this.props.replaceDirtyChannel(new Map({name: this.state.channelName, isDirty: true}));
+    const hasError = this.validateChannelName(channelName);
+    if (!hasError) {
+      this.props.replaceDirtyChannel(new Map({name: channelName, isDirty: true}));
       this.props.newChannel({name: channelName});
       const newChannelItem = this.refs.newChannel.getDOMNode();
       newChannelItem.className = newChannelItem.className + ' new-channel_disabled';
```
Comment (Nitive) [inline app/components/NewChannelItem/index.jsx]: Ups, we have classname module for this

Comment (Nitive) [inline app/components/NewChannelItem/index.jsx]: What the angular in our code?
I thought we correct that

Comment (Faraser) [inline app/components/NewChannelItem/index.jsx]: Where did you see the angular in this code? I run a validate function and check results. If there are no errors i run a addChannel action.

Comment (Nitive) [inline app/components/NewChannelItem/index.jsx]: I am about old code in lines 54-55

Comment (Faraser) [inline app/components/NewChannelItem/index.jsx]: Oh, I do not understand you correctly. Yes it is:)

**Pull request merged**